### ISSUE TITLE
Deprecate and remove MarkV and Stats tools?

### ISSUE THREAD

**Issue by dj2** [KhronosGroup/SPIRV-Tools#1839]
Neither of these tools are under active development. I propose we mark them as deprecated by printing to the console on usage. Give a deadline of ~6 months after we add the warning, and then remove the tools.

Is there a good use-case for these that I'm missing? There is a SPIRV SMOL that does compression already and I'm not sure how useful the stats code is in general.

**Comment by dj2** [comment]
@dneto0 thoughts, concerns?

**Comment by aras-p** [comment]
> There is a SPIRV SMOL that does compression already

_(author of SMOL-V here)_

I just remembered that MARK-V is here, and thus went to compare it with SMOL-V. So far my impression is:

- MARK-V by itself is not strong; e.g. SMOL-V + LZ4 (or any of the stronger general lossless compressors) always beats it on my test corpus of various SPIR-V shaders.
- However, MARK-V with additional general lossless compression does tend to produce slightly smaller results compared to SMOL-V with the same lossless compression on top.
- If someone needs MARK-V in their application to decompress shaders at load time, they are looking at 1.9MB (!) of executable code. So far not 100% sure I did everything correctly, but it seems like it. In comparison, SMOL-V decoder is about 6 kilobytes.
- Hit some asserts when decompressing some shaders compressed by MARK-V, will report issues here soon.
